plugins/CoreHome/Columns/Metrics/EvolutionMetric.php

```diff
@@ -170,4 +179,40 @@ private function getPastDataTable()
         }
         return $result;
     }
+
+    /**
+     * Calculate the ratio of time between a past period and current incomplete period
+     *
+     * eg. if today is Thursday at 12:00pm and the past period is a week then the ratio is 0.5, exactly half of the
+     * current incomplete period has passed
+     *
+     * If the current period end is in the past then the ratio will always be 1, since the current period is complete.
+     *
+     * @param DataTable $currentData
+     * @param DataTable $pastData
+     * @return float|int
+     * @throws \Exception
+     */
+    public static function getRatio(DataTable $currentData, DataTable $pastData)
+    {
+        $ratio = 1;
+
+        $p = $pastData->getMetadata('period');
+        $pStart = new \DateTime($p->getDateStart().' 00:00:00');
+        $pEnd = new \DateTime($p->getDateEnd().' 23:59:59');
+
+        $c = $currentData->getMetadata('period');
+        $cStart = new \DateTime($c->getDateStart().' 00:00:00');
+        $cEnd = new \DateTime($c->getDateEnd().' 23:59:59');
+
+        $now = new \DateTime("now");
```

Comment:
This may not be 100% accurate but we can possibly ignore. In some other data tables we have the information when the report was last generated as a metadata in `DataTable::ARCHIVED_DATE_METADATA_NAME`. That might be interesting because certain periods we might only archive every 6 hours or once per day meaning the math would be a bit off. Eg yearly, monthly we often only calculate once a day.

I wonder if somewhere in the MultiSite report generation this metadata gets lost or whether it's never there in the first place. If we could use the date when the report was generated that could make quite a difference.
![image](https://user-images.githubusercontent.com/273120/136109020-4d55c296-cd78-473b-b796-60ac727f4550.png)